src/diem/offchain/funds_pull_pre_approval_command_state.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import typing
+
+from .types import FundPullPreApprovalStatus
+
+Status = str
+
+_transitions: typing.Dict[Status, typing.List[Status]] = {
+    FundPullPreApprovalStatus.pending: [
+        FundPullPreApprovalStatus.pending,
+        FundPullPreApprovalStatus.valid,
+        FundPullPreApprovalStatus.rejected,
+        FundPullPreApprovalStatus.closed,
+    ],
+    FundPullPreApprovalStatus.valid: [
+        FundPullPreApprovalStatus.valid,
+        FundPullPreApprovalStatus.closed,
+    ],
+    FundPullPreApprovalStatus.rejected: [
+        FundPullPreApprovalStatus.rejected,
```

Comment:
I don't think this should be allowed, after rejected, it is end

Reply:
This is allowed due to the idempotency, as described above.